rust/connection-tests/docker-compose.wan-relay.yml

```diff
@@ -0,0 +1,154 @@
+# This test environment partitions dialer and listener into different subnets.
+# Their router is configured to use `fully-random` port mapping, making hole-punching impossible.
+
+version: "3.8"
+name: wan-relay-integration-test
+
+services:
+  dialer:
+    build:
+      target: debug
+      context: ..
+      args:
+        PACKAGE: firezone-connection-tests
+      cache_from:
+        - type=registry,ref=us-east1-docker.pkg.dev/firezone-staging/cache/connection-tests:main
+    image: us-east1-docker.pkg.dev/firezone-staging/firezone/connection-tests:${VERSION:-main}
+    environment:
+      ROLE: "dialer"
+    cap_add:
+      - NET_ADMIN
+    entrypoint: /bin/sh
+    command:
+      - -c
+      - |
+        set -ex
+
+        ROUTER_IP=$$(dig +short dialer_router)
+        INTERNET_SUBNET=$$(curl --fail --silent --unix-socket /var/run/docker.sock http://localhost/networks/wan-relay-integration-test_wan | jq -r '.IPAM.Config[0].Subnet')
+
+        ip route add $$INTERNET_SUBNET via $$ROUTER_IP dev eth0
+
+        export TURN_SERVER=$$(curl --fail --silent --unix-socket /var/run/docker.sock http://localhost/containers/wan-relay-integration-test-relay-1/json | jq -r '.NetworkSettings.Networks."wan-relay-integration-test_wan".IPAddress')
```

Comment:
Note for myself, check if we can use `dig` here if we explicitly provide the containers name in the config.